src/overlays/actors/ovl_En_Holl/z_en_holl.c

```diff
@@ -152,162 +213,186 @@ void func_80A58DD4(EnHoll* this, PlayState* play) {
     }
 }
 
-// Horizontal Planes
-void func_80A59014(EnHoll* this, PlayState* play) {
+void EnHoll_HorizontalInvisible(EnHoll* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
     s32 useViewEye = gDbgCamEnabled || play->csCtx.state != CS_STATE_IDLE;
-    Vec3f vec;
-    s32 temp;
-    f32 planeHalfWidth;
-    f32 absZ;
-
-    func_8002DBD0(&this->actor, &vec, (useViewEye) ? &play->view.eye : &player->actor.world.pos);
-    planeHalfWidth = (((this->actor.params >> 6) & 7) == 6) ? PLANE_HALFWIDTH : PLANE_HALFWIDTH_2;
-
-    temp = EnHoll_IsKokiriLayer8();
-    if (temp || (PLANE_Y_MIN < vec.y && vec.y < PLANE_Y_MAX && fabsf(vec.x) < planeHalfWidth &&
-                 (absZ = fabsf(vec.z), 100.0f > absZ && absZ > 50.0f))) {
-        s32 transitionActorIdx = GET_TRANSITION_ACTOR_INDEX(&this->actor);
-        s32 side = (vec.z < 0.0f) ? 0 : 1;
-        TransitionActorEntry* transitionEntry = &play->transiActorCtx.list[transitionActorIdx];
+    Vec3f relSubjectPos;
+    s32 isKokiriLayer8;
+    f32 hollHalfWidth;
+    f32 orthogonalDistToSubject;
+
+    func_8002DBD0(&this->actor, &relSubjectPos, useViewEye ? &play->view.eye : &player->actor.world.pos);
+    hollHalfWidth =
+        (ENHOLL_GET_TYPE(&this->actor) == ENHOLL_H_INVISIBLE_NARROW) ? ENHOLL_H_HALFWIDTH_NARROW : ENHOLL_H_HALFWIDTH;
+
+    isKokiriLayer8 = EnHoll_IsKokiriLayer8();
```

Comment:
While you're here, I'm looking at the function:
```c
s32 EnHoll_IsKokiriLayer8(void) {
    return gSaveContext.entranceIndex == ENTR_SPOT04_0 && gSaveContext.sceneLayer == 8;
}
```
And Fig's Cutscene PR:
```c
case CS_DEST_KOKIRI_FOREST_FROM_BLUE_WARP:
    play->nextEntranceIndex = ENTR_SPOT04_0;
    gSaveContext.cutsceneIndex = 0xFFF8;
    play->transitionTrigger = TRANS_TRIGGER_START;
    play->transitionType = TRANS_TYPE_FADE_WHITE;
    gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE;
    break;
```
Methinks `EnHoll_IsKokiriLayer8` can be named to something like `EnHoll_IsKokiriForestEntranceFromBlueWarp` or something similar.

Reply:
Oh wait, did I mess this up, it might need `0xFFF4` for the cutsceneIndex, brb

Reply:
this is cutscene 8 which is layer 12:

0 child day
1 child night
2 adult day
3 adult night
4 cutscene 0
...
12 cutscene 8

Reply:
Thanks, brain not functioning tonight

Reply:
Nice try though :sunglasses: